### ISSUE TITLE
Track successful code execution with smbrelayx and stop on success

### ISSUE THREAD

**Issue by ropnop** [CoreSecurity/impacket#168]
Creating a new issue after the discussion here:

https://github.com/CoreSecurity/impacket/issues/132

The new `-one-shot` option in smbrelayx stops after successful authentication, not successful code execution (aka it stops whether or not the relayed hash has admin privileges).

The use-case would be during a pentest to let Responder and smbrelayx run continuously to capture hashes. If you get "lucky" and a captured hash is a local admin on the target, the payload executes and you can stop.

One issue to figure out is how `doAttack()` can let the HTTP/SMB server know it succeeded so they can add the host to the `attacked` set. Currently with threading it is difficult to return a value in time before the next request comes in. Not sure the best way to go about doing this just yet.

**Comment by asolino** [comment]
hey @ropnop, can you please give it a try and check whether this solve the issue?

thanks!

**Comment by ropnop** [comment]
Confirmed! Nice work. Scenario tested:

I set up Responder and ran the follwoing command:
```
python smbrelayx.py -h 10.9.122.10 -one-shot
```

First, user `kbryant` was LLMNR spoofed and redirected to `smbrelayx.py`. The relay happened but got `[-] DCERPC Runtime Error: code: 0x5 - rpc_s_access_denied` because he is not a local admin.

Then, user `jhoyer` was spoofed and his creds were relayed. He was an admin so `secretsdump` was run succesfully.

A few seconds later, `jhoyer` was spoofed again. This time the tool responds with: 
`[*] HTTPD: Received connection from 10.9.122.5, skipping 10.9.122.10, already attacked`.

As a final test, another third user, `jarrieta` was spoofed. The tool responded appropriately with "already attacked".

I haven't tested it for speed yet (e.g. if another spoofed request comes in before `secretsdump` finishes), but that's an edge case and it's working exactly how I wanted it. 

Thanks! This will be super useful on my next internal pentest! :)
